src/client/java/minicraft/screen/GameToast.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Minicraft+ contributors
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package minicraft.screen;
+
+import minicraft.gfx.Dimension;
+import minicraft.gfx.Screen;
+import minicraft.gfx.Sprite;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * All implementations should all be based on sprites without custom drawing function invoke.
+ * Dynamic dimensions are not supported, and sizes should maintain static.
+ * Game Toasts are rendered only in gameplay and always located in the top-left corner without margin.
+ * The content of game toasts is implementation-specific.
+ */
+public abstract class GameToast extends Toast {
```

Comment:
Why did you add this class??

Reply:
To constrain and categorize to have only game toasts and app toasts more exactly. In design, it is not expected to see app toasts being in the list of game toasts, and game toasts are expected to behave similarly, different from app toasts in certain aspects.

Reply:
We already discussed this. You removed it. And now you're adding it back? Adding it is just a way of writing more code that does less. Adding code for the sake of adding code is pointless.

Reply:
I have not removed it, but ignored it from the implementation to simplify the structure (in fact it should be specialized for the API). Also, adding design constrains preventing unexpected behaviors is an important thing for designing API.

Reply:
Good this is a game and not an API then